cloud-reactor/src/main/java/com/sequenceiq/cloudbreak/cloud/handler/StopStartUpscaleStartInstancesHandler.java

```diff
@@ -114,7 +123,25 @@ public void accept(Event<StopStartUpscaleStartInstancesRequest> event) {
             CloudConnector<?> connector, AuthenticatedContext ac, List<CloudInstance> instancesToStart) {
         // Note: This timebound can apply thrice along with some delays, given how retries on this API are currently configured for AWS.
         //  The rest of the cloud providers is an UNKNOWN.
-        return connector.instances().startWithLimitedRetry(ac, null, instancesToStart, START_POLL_TIMEBOUND_MS);
+        try {
+            return connector.instances().startWithLimitedRetry(ac, null, instancesToStart, START_POLL_TIMEBOUND_MS);
+        } catch (Exception originalException) {
+            // We exceeded the time-bound on the polling. Try getting the status for all the instances that we attempted to start.
+            if (originalException instanceof PollerStoppedException) {
+                LOGGER.warn("Timed out while attempting to start instances. Attempting to get states for attempted nodes", originalException);
+            } else {
+                LOGGER.warn("Exception while attempting to start instances. Attempting to get states for attempted nodes", originalException);
+            }
+            try {
+                return connector.instances().checkWithoutRetry(ac, instancesToStart);
+            } catch (Exception e) {
+                LOGGER.warn("Error while trying to get instance status after start failure. Propagating original error with the start attempt", e);
+                // The message in case of PollerStoppedException is far too verbose for the console.
+                String message = originalException instanceof PollerStoppedException ? "Timed out while waiting for instances to start" :
```

Comment:
what if the new exception indicates an actual error like throttling or network issue or permission issue? then we will hide it if the original exception was a timeout

Reply:
The new exception gets logged. The original exception is propagated since that is the actual failure. (This operation is an attempt to recover from that).